The test is failing, so this can't be merged. This explains why the `no_oss` tag was there: The test fails, and since the timeout of the test is long, it doesn't run on the internal codebase either, so the depthwise convolution tests are not run at all by the CI.

The tests `testDepthwiseConv2DFilterGradCompare` and `testDepthwiseConv2DFilterGradExplicitCompare` are failing, both on the CPU and GPU builds. The former test fails with the error

```
======================================================================
ERROR: testDepthwiseConv2DFilterGradCompare (tensorflow.python.kernel_tests.nn_ops.depthwise_conv_op_base.DepthwiseConv2DBase)
tensorflow.python.kernel_tests.nn_ops.depthwise_conv_op_base.DepthwiseConv2DBase.testDepthwiseConv2DFilterGradCompare
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/b/f/w/bazel-out/k8-opt/bin/tensorflow/python/kernel_tests/nn_ops/depthwise_conv_op_test_gpu.runfiles/org_tensorflow/tensorflow/python/kernel_tests/nn_ops/depthwise_conv_op_base.py", line 1055, in testDepthwiseConv2DFilterGradCompare
    self._CompareBackpropFilter(input_size, filter_size, output_size, stride,
  File "/b/f/w/bazel-out/k8-opt/bin/tensorflow/python/kernel_tests/nn_ops/depthwise_conv_op_test_gpu.runfiles/org_tensorflow/tensorflow/python/kernel_tests/nn_ops/depthwise_conv_op_base.py", line 1043, in _CompareBackpropFilter
    gpu_value = _GetVal(use_gpu=True, data_format=data_format)
  File "/b/f/w/bazel-out/k8-opt/bin/tensorflow/python/kernel_tests/nn_ops/depthwise_conv_op_test_gpu.runfiles/org_tensorflow/tensorflow/python/kernel_tests/nn_ops/depthwise_conv_op_base.py", line 1035, in _GetVal
    backprop = nn_ops.depthwise_conv2d_native_backprop_filter(
  File "/b/f/w/bazel-out/k8-opt/bin/tensorflow/python/kernel_tests/nn_ops/depthwise_conv_op_test_gpu.runfiles/org_tensorflow/tensorflow/python/util/traceback_utils.py", line 141, in error_handler
    return fn(*args, **kwargs)
  File "/b/f/w/bazel-out/k8-opt/bin/tensorflow/python/kernel_tests/nn_ops/depthwise_conv_op_test_gpu.runfiles/org_tensorflow/tensorflow/python/util/dispatch.py", line 1170, in op_dispatch_handler
    return dispatch_target(*args, **kwargs)
  File "/b/f/w/bazel-out/k8-opt/bin/tensorflow/python/kernel_tests/nn_ops/depthwise_conv_op_test_gpu.runfiles/org_tensorflow/tensorflow/python/ops/nn_ops.py", line 3150, in depthwise_conv2d_native_backprop_filter
    return gen_nn_ops.depthwise_conv2d_native_backprop_filter(
  File "/b/f/w/bazel-out/k8-opt/bin/tensorflow/python/kernel_tests/nn_ops/depthwise_conv_op_test_gpu.runfiles/org_tensorflow/tensorflow/python/ops/gen_nn_ops.py", line 2540, in depthwise_conv2d_native_backprop_filter
    _ops.raise_from_not_ok_status(e, name)
  File "/b/f/w/bazel-out/k8-opt/bin/tensorflow/python/kernel_tests/nn_ops/depthwise_conv_op_test_gpu.runfiles/org_tensorflow/tensorflow/python/framework/ops.py", line 7203, in raise_from_not_ok_status
    raise core._status_to_exception(e) from None  # pylint: disable=protected-access
tensorflow.python.framework.errors_impl.InvalidArgumentError: {{function_node __wrapped__DepthwiseConv2dNativeBackpropFilter_device_/job:[localhost/replica:0](https://cs.corp.google.com/piper///depot/google3/localhost/replica?cl=446264337)/task:0/device:GPU:0}} DepthwiseConv2DBackpropFilter: input and filter must have the same in_depth [Op:DepthwiseConv2dNativeBackpropFilter]
```

The latter test fails with

```
======================================================================
ERROR: testDepthwiseConv2DFilterGradExplicitCompare (tensorflow.python.kernel_tests.nn_ops.depthwise_conv_op_base.DepthwiseConv2DBase)
tensorflow.python.kernel_tests.nn_ops.depthwise_conv_op_base.DepthwiseConv2DBase.testDepthwiseConv2DFilterGradExplicitCompare
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/b/f/w/bazel-out/k8-opt/bin/tensorflow/python/kernel_tests/nn_ops/depthwise_conv_op_test_gpu.runfiles/org_tensorflow/tensorflow/python/kernel_tests/nn_ops/depthwise_conv_op_base.py", line 1073, in testDepthwiseConv2DFilterGradExplicitCompare
    self._CompareBackpropFilter(input_size, filter_size, output_size, stride,
  File "/b/f/w/bazel-out/k8-opt/bin/tensorflow/python/kernel_tests/nn_ops/depthwise_conv_op_test_gpu.runfiles/org_tensorflow/tensorflow/python/kernel_tests/nn_ops/depthwise_conv_op_base.py", line 1043, in _CompareBackpropFilter
    gpu_value = _GetVal(use_gpu=True, data_format=data_format)
  File "/b/f/w/bazel-out/k8-opt/bin/tensorflow/python/kernel_tests/nn_ops/depthwise_conv_op_test_gpu.runfiles/org_tensorflow/tensorflow/python/kernel_tests/nn_ops/depthwise_conv_op_base.py", line 1035, in _GetVal
    backprop = nn_ops.depthwise_conv2d_native_backprop_filter(
  File "/b/f/w/bazel-out/k8-opt/bin/tensorflow/python/kernel_tests/nn_ops/depthwise_conv_op_test_gpu.runfiles/org_tensorflow/tensorflow/python/util/traceback_utils.py", line 141, in error_handler
    return fn(*args, **kwargs)
  File "/b/f/w/bazel-out/k8-opt/bin/tensorflow/python/kernel_tests/nn_ops/depthwise_conv_op_test_gpu.runfiles/org_tensorflow/tensorflow/python/util/dispatch.py", line 1170, in op_dispatch_handler
    return dispatch_target(*args, **kwargs)
  File "/b/f/w/bazel-out/k8-opt/bin/tensorflow/python/kernel_tests/nn_ops/depthwise_conv_op_test_gpu.runfiles/org_tensorflow/tensorflow/python/ops/nn_ops.py", line 3150, in depthwise_conv2d_native_backprop_filter
    return gen_nn_ops.depthwise_conv2d_native_backprop_filter(
  File "/b/f/w/bazel-out/k8-opt/bin/tensorflow/python/kernel_tests/nn_ops/depthwise_conv_op_test_gpu.runfiles/org_tensorflow/tensorflow/python/ops/gen_nn_ops.py", line 2540, in depthwise_conv2d_native_backprop_filter
    _ops.raise_from_not_ok_status(e, name)
  File "/b/f/w/bazel-out/k8-opt/bin/tensorflow/python/kernel_tests/nn_ops/depthwise_conv_op_test_gpu.runfiles/org_tensorflow/tensorflow/python/framework/ops.py", line 7203, in raise_from_not_ok_status
    raise core._status_to_exception(e) from None  # pylint: disable=protected-access
tensorflow.python.framework.errors_impl.InvalidArgumentError: {{function_node __wrapped__DepthwiseConv2dNativeBackpropFilter_device_/job:[localhost/replica:0](https://cs.corp.google.com/piper///depot/google3/localhost/replica?cl=446264337)/task:0/device:GPU:0}} Nonzero explicit padding in the batch or depth dimensions is not supported [Op:DepthwiseConv2dNativeBackpropFilter]
```

This needs to get fixed, as we are not testing depthwise convolutions currently. @duncanriach, do you want to fix this? A quick fix would be to add `@unittest.skip("Test currently fails")` for the two failing test methods.

--- a/tensorflow/python/kernel_tests/nn_ops/BUILD
+++ b/tensorflow/python/kernel_tests/nn_ops/BUILD
@@ -331,10 +331,6 @@ cuda_py_test(
     timeout = "long",
     srcs = ["depthwise_conv_op_d9m_test.py"],
     shard_count = 8,
-    tags = [
-        "no_oss",  # TODO(b/118842098): Re-enable this test in Kokoro.
-        "notap",  # TODO(b/231615784): Re-enable this test once fixed.
-    ],
     deps = [
         ":depthwise_conv_op_base",
         "//tensorflow/python:random_ops",
@@ -348,10 +344,6 @@ cuda_py_test(
     timeout = "long",
     srcs = ["depthwise_conv_op_test.py"],
     shard_count = 8,
-    tags = [
-        "no_oss",  # TODO(b/118842098): Re-enable this test in Kokoro.
-        "notap",  # TODO(b/231615784): Re-enable this test once fixed.
-    ],
     deps = [
         ":depthwise_conv_op_base",
     ],

--- a/tensorflow/python/kernel_tests/nn_ops/depthwise_conv_op_base.py
+++ b/tensorflow/python/kernel_tests/nn_ops/depthwise_conv_op_base.py
@@ -974,6 +974,7 @@ class DepthwiseConv2DBase(test.TestCase):
     cpu_value = _GetVal(use_gpu=False)
     self.assertAllClose(cpu_value, gpu_value, rtol=1e-4, atol=1e-4)
 
+  @test_util.run_gpu_only
   def testDepthwiseConv2DInputGradCompare(self):
     for index, (input_size, filter_size, output_size, stride, padding,
                 dilations) in enumerate(ConfigsToTest()):
@@ -992,6 +993,7 @@ class DepthwiseConv2DBase(test.TestCase):
       self._CompareBackpropInput(input_size, filter_size, output_size, stride,
                                  padding, "float64")
 
+  @test_util.run_gpu_only
   def testDepthwiseConv2DInputGradExplicitCompare(self):
     for index, (input_size, filter_size, output_size, stride, padding,
                 dilations) in enumerate(ConfigsToTestExplicit()):
@@ -1044,6 +1046,7 @@ class DepthwiseConv2DBase(test.TestCase):
       gpu_value = _GetVal(use_gpu=True, data_format=data_format)
       self.assertAllClose(cpu_value, gpu_value, rtol=1e-4, atol=1e-4)
 
+  @test_util.run_gpu_only
   def testDepthwiseConv2DFilterGradCompare(self):
     for index, (input_size, filter_size, output_size, stride, padding,
                 dilations) in enumerate(ConfigsToTest()):
@@ -1062,6 +1065,7 @@ class DepthwiseConv2DBase(test.TestCase):
       self._CompareBackpropFilter(input_size, filter_size, output_size, stride,
                                   padding, "float64")
 
+  @test_util.run_gpu_only
   def testDepthwiseConv2DFilterGradExplicitCompare(self):
     for index, (input_size, filter_size, output_size, stride, padding,
                 dilations) in enumerate(ConfigsToTestExplicit()):

--- a/tensorflow/python/kernel_tests/nn_ops/depthwise_conv_op_base.py
+++ b/tensorflow/python/kernel_tests/nn_ops/depthwise_conv_op_base.py
@@ -1033,7 +1033,8 @@ class DepthwiseConv2DBase(test.TestCase):
           strides = [1, 1, stride, stride]
           padding = padding_nchw
         backprop = nn_ops.depthwise_conv2d_native_backprop_filter(
-            t0, t1, t2, strides=strides, padding=padding)
+            t0, t1, t2, strides=strides, padding=padding,
+            data_format=data_format)
         ret = self.evaluate(backprop)
         self.assertShapeEqual(ret, backprop)
         return ret

